src/System.Private.ServiceModel/tests/Common/Unit/TestHelpers.cs

```diff
@@ -276,3 +277,69 @@ protected override void OnWriteBodyContents(XmlDictionaryWriter writer)
         writer.WriteString(_bodyContent);
     }
 }
+
+public class WcfDuplexServiceCallback : IWcfDuplexServiceCallback
+{
+    public Task<Guid> OnPingCallback(Guid guid)
+    {
+        return Task.FromResult<Guid>(guid);
+    }
+
+    void IWcfDuplexServiceCallback.OnPingCallback(Guid guid)
+    {
+
+    }
+}
+
+public class MyOperationBehavior : Attribute, IOperationBehavior
+{
+    public static StringBuilder errorBuilder = new StringBuilder();
+    public static Collection<string> collection = new Collection<string>();
+
+    // Pass data at runtime to bindings to support custom behavior
+    public void AddBindingParameters(OperationDescription operationDescription, BindingParameterCollection bindingParameters)
+    {
+        collection.Add("calledAddBindingParameters");
+
+        if (operationDescription == null)
+        {
+            errorBuilder.AppendLine(String.Format("A parameter passed into the AddBindingParameters method was null/nThe null parameter is: {0}", typeof(OperationDescription).ToString()));
+        }
+        if (bindingParameters == null)
+        {
+            errorBuilder.AppendLine(String.Format("A parameter passed into the AddBindingParameters method was null/nThe null parameter is: {0}", typeof(BindingParameterCollection).ToString()));
+        }
+    }
+
+    // Implements a modification or extension of the client across an operation
+    public void ApplyClientBehavior(OperationDescription operationDescription, ClientOperation clientOperation)
+    {
+        collection.Add("calledApplyClientBehavior");
+
+        if (operationDescription == null)
+        {
+            errorBuilder.AppendLine(String.Format("A parameter passed into the ApplyClientBehavior method was null/nThe null parameter is: {0}", typeof(OperationDescription).ToString()));
+        }
+        if (clientOperation == null)
+        {
+            errorBuilder.AppendLine(String.Format("A parameter passed into the ApplyClientBehavior method was null/nThe null parameter is: {0}", typeof(ClientOperation).ToString()));
+        }
+    }
+
+    // Implements a modification or extension of the service across an operation
+    public void ApplyDispatchBehavior(OperationDescription operationDescription, DispatchOperation dispatchOperation)
+    {
+        // This method does not get called client side.
+    }
```

Comment:
Optional: it's not *supposed* to get called, but you could make the unit test also handle that negative case by writing to the errorBuilder "ApplyDispatchBehavior should not have been called".